### DISCUSSION TITLE
Creality 4.2.2 Board 2nd serial port ?

### DISCUSSION THREAD

**Discussion by barbudor** [luc-github/ESP3D#557] [Questions]
Hi

I'd like to add ESP3D to my Creality Ender 3 based on board 4.2.2

I can't find information on where a 2nd serial port could be available (I understand 1st port is connected to the CH340 USB/serial adapter). 

Thanks for any pointer

**Comment by luc-github** [comment]
the only info here is https://github.com/luc-github/ESP3D/issues/524#issuecomment-745163565

**Reply by barbudor** [reply]
Thanks Luc

However I understand this connects the ESP in parallel of the CH340 with 2 TX pins (one of the CH + one of the ESP) pusging on the same line.
I wonder about the risk of damaging either the CH340 or the ESP. And may prevent using the USB in the future.

My understanding is that there should be another serial port available that could be enabled in Marlin with `#define SERIAL_PORT_2 3`
But can't find any information on the hardware of the board.

Merci 👍

**Comment by luc-github** [comment]
well even the chip has several serial ports, the pins may be used  for another purpose already - the solution is to check the chip specs of your board - locate the serial pins on mcu and follow the traces to see what use them and if you can use them

**Comment by tino-q** [comment]
@barbudor bump.

Did you manage any progress on figuring out 2nd serial port pins?

**Reply by barbudor** [reply]
Hi @tino-q 
No. I didn't looked further.

**Reply by samir-101** [reply]
The second serial TX pin PA2 is used with heater bed i don't think you can use that!

**Comment by marklouie182** [comment]
Hello Sirs... Is it the safest pin (CH340 RX TX) to use for ESP3D to interface NodeMCU? Using v4.2.2 board...
Thank you very much...

**Reply by barbudor** [reply]
Yes,
AS far as you don't plug anything in the USB, the CH340 pins are high-impedance so won't disturb the communucation between the ST MCU and the ESP on the NodeMCU.
I have soldered RX+RX directly n the CH340, and I pick GND and +5V on the BLTouch connector.
